### DISCUSSION TITLE
AddressBook suggestions

### DISCUSSION THREAD

**Discussion by it-boerner** [rustdesk/rustdesk#5024] [Feature Request]
I logged into the RustDesk client on two different computers. The clients that have been created are displayed, but some of the saved names are not adopted and the saved passwords in client 1 are not made available to client 2 either.

**Comment by rustdesk** [comment]
Sorry, not get you this time.

**Comment by it-boerner** [comment]
I installed RustDesk on Computer 1. There I logged into the client and made all my address book entries.

I also saved a password for each client.

If I now log on to computer 2 with the same user name, all the address book entries are displayed, but some of the names are not all displayed correctly. Also, the saved passwords from computer 1 are not present on computer 2. It will be asked for the password each time a new connection is made.

**Comment by rustdesk** [comment]
The address book does not save password, the other vendors have this behavior too.
@21pages does address book save hostname and username?

**Reply by 21pages** [reply]
not save now

**Comment by it-boerner** [comment]
Your statement is not entirely correct. If you have added a host in TeamViewer, it will be saved in your address book including the password. It then doesn't matter on which computer you log in with your user data, because the data is also available in the other TeamViewer clients.

Is there another way to transfer the passwords? Maybe a copy of the folder?

**Comment by rustdesk** [comment]
@21pages which vendor does not save password?

**Reply by 21pages** [reply]
anydesk

**Comment by it-boerner** [comment]
Please consider, RustDesk has the potential to become big and a good competitor to TeamViewer and not to AnyDesk.

If you have a lot of clients and work on different computers, you cannot always remember the passwords.

It should be possible to store passwords centrally.

**Reply by rustdesk** [reply]
You are right. Without rememering password, the addressbook seems useless. Also hostname, not sure about username.

**Reply by it-boerner** [reply]
Thank you for sharing my view.

I really want to support you in your project and make good suggestions for improvement.

I've been working with TeamViewer for a few years and the pricing policy is a disaster.

We should create a reasonable address book that also works logically for the end user.

**Reply by rustdesk** [reply]
Can you show me what names displayed on TV addressbook?

**Reply by it-boerner** [reply]
A two-factor authentication should be installed, so after entering the username and password, the 2FA would be required and the data would be stored correctly and securely behind the username.

**Reply by rustdesk** [reply]
2FA is in our plan.

**Reply by it-boerner** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/118393600/ae136ca5-3818-40c6-8662-e2653a1ffa97)

**Reply by it-boerner** [reply]
By integrating the 2FA, the data would also be secured in the database, so I can access my data (like clients and passwords) with my username, password and the 2FA.

**Reply by rustdesk** [reply]
The different names are alias?

**Reply by dinger1986** [reply]
> Can you show me what names displayed on TV addressbook?

I can show you my teamviewer at some point if you would like?

**Reply by it-boerner** [reply]
No, it´s the Name

**Reply by it-boerner** [reply]
When I click on the ID and want to rename it, it gives me the correct name.

**Reply by rustdesk** [reply]
So you renamed, but the new name is not displayed?

**Reply by it-boerner** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/118393600/73724412-4322-4702-9291-04cd80eadadc)

A screenshot of TeamViewer.

Data such as ID, password, alias name and the group are saved in TeamViewer.

No matter where I log in with my TeamViewer username, I get the same data.

**Reply by rustdesk** [reply]
Thanks for your feedback, we will polish it ASAP

**Reply by it-boerner** [reply]
You are great and I am happy

**Comment by it-boerner** [comment]
>So you renamed, but the new name is not displayed?

Yes thats right.

**Comment by rustdesk** [comment]
https://community.teamviewer.com/English/kb/articles/108791-two-factor-authentication-for-connections

**Reply by it-boerner** [reply]
If you want to connect to a client you need the password or the 2FA. That's the idea behind it, isn't it?

**Reply by dinger1986** [reply]
To login at all you should need 2fa, even if it's just an email with a code or authorisation button, (but that could be sent by the api server and switched on or off from there in the web console) obviously until emails are setup then that will or won't work anyway :)

**Reply by it-boerner** [reply]
I think it sounds like a good plan. But one should also consider using a smartphone to use 2FA. We want to stay modern :)

**Reply by dinger1986** [reply]
Yes could do it by code so mfa app or bitwarden or whatever or email.

Anything's better than nothing
